packages/plugins/typescript/effect/src/visitor.ts

```diff
@@ -0,0 +1,274 @@
+import autoBind from 'auto-bind';
+import { type GraphQLSchema, type OperationDefinitionNode, print } from 'graphql';
+import {
+  type ClientSideBasePluginConfig,
+  ClientSideBaseVisitor,
+  DocumentMode,
+  getConfigValue,
+  LoadedFragment,
+} from '@graphql-codegen/visitor-plugin-common';
+import type { RawEffectPluginConfig } from './config.js';
+
+export type EffectPluginConfig = ClientSideBasePluginConfig &
+  (
+    | {
+        mode?: 'client-only' | 'mixed';
+      }
+    | {
+        mode: 'operations-only';
+        relativeClientImportPath: string;
+      }
+  );
+
+const clientCode = `export type GraphQLSuccessResponse<A = any> = Pick<
+  Http.response.ClientResponse,
+  'status' | 'headers'
+> & {
+  readonly body: ExecutionResult & { readonly data: A };
+};
+
+export type GraphQLErrorResponse = Pick<Http.response.ClientResponse, 'status' | 'headers'> & {
+  readonly body: Pick<ExecutionResult, 'errors' | 'extensions'>;
+};
+
+export class MissingDataGraphQLResponseError extends Data.TaggedError(
+  'MissingDataGraphQLResponseError',
+)<GraphQLErrorResponse> {}
+
+const headers = {
+  // https://github.com/graphql/graphql-over-http/blob/main/spec/GraphQLOverHTTP.md#legacy-watershed
+  Accept: 'application/graphql-response+json; charset=utf-8, application/json; charset=utf-8',
+  'Content-Type': 'application/json',
+};
+
+export class GraphQLClient extends Context.Tag('GraphQLClient')<
+  GraphQLClient,
+  Http.client.Client<
```

Comment:
Shouldn't this be `<GraphQLSuccessResponse, Http.error.HttpClientError | MissingDataGraphQLResponseError, never>` now?

Reply:
I haven't upgraded to the latest version of effect, yet. I'm first trying to figure out what's wrong with the current build. Yarn's dependency hoisting got messed up after I added this new workspace. I don't understand what's going on and how to work around the issue.